default/python/AI/fleet_orders.py

```diff
@@ -27,24 +28,35 @@ def trooper_move_reqs_met(main_fleet_mission, order, verbose):
     # is already a military fleet assigned to secure the target, and don't take final jump unless the planet is
     # (to the AI's knowledge) down to zero shields.  Additional checks will also be done by the later
     # generic movement code
-    system_id = order.fleet.get_system().id
     invasion_target = main_fleet_mission.target
     invasion_planet = invasion_target.get_object()
     invasion_system = invasion_target.get_system()
     supplied_systems = fo.getEmpire().fleetSupplyableSystemIDs
     # if about to leave supply lines
-    if (order.target.id not in supplied_systems and system_id in supplied_systems):
-        if (invasion_planet.currentMeterValue(fo.meterType.maxShield) and
-                    invasion_system.id not in AIstate.militarySystemIDs):
-            if verbose:
-                print ("trooper_move_reqs_met() holding Invasion fleet %d before leaving supply "
-                       "because target (%s) has nonzero max shields and there is not yet a military fleet "
-                       "assigned to secure the target system.") % (order.fleet.id, invasion_planet)
-            return False
+    if order.target.id not in supplied_systems:
+        if invasion_planet.currentMeterValue(fo.meterType.maxShield):
+            military_support_fleets = MilitaryAI.get_military_fleets_with_target_system(invasion_system.id)
+            if not military_support_fleets:
+                if verbose:
+                    print ("trooper_move_reqs_met() holding Invasion fleet %d before leaving supply "
+                           "because target (%s) has nonzero max shields and there is not yet a military fleet "
+                           "assigned to secure the target system.") % (order.fleet.id, invasion_planet)
+                return False
+
+            eta_this_fleet = FleetUtilsAI.calculate_estimated_time_of_arrival(order.fleet.id, invasion_system.id)
+            if all((eta_this_fleet - 1) < FleetUtilsAI.calculate_estimated_time_of_arrival(fid, invasion_system.id)
```

Comment:
Seems to me like this should either drop the -1 or make the comparison be <=.

Also, just noting, I expect that at some point in the future we'd want to be able to allow, under certain circumstances, for the troopers and warfleet to arrive on the same turn.  Perhaps add a todo comment for that?